lib/ruboty/slack_socket_mode/interactive_message.rb

```diff
@@ -46,6 +46,37 @@ def delete
 
         robot.delete_interactive(response_url)
       end
+
+      def update(text: nil, block: nil)
+        response_url = @payload['response_url']
+        unless response_url
+          Ruboty.logger.warn("#{self.class.name}: Cannot update message. This message does not contain response_url in payload.")
+          return
+        end
+
+        if text.nil? && block.nil? || text && block
+          Ruboty.logger.warn("#{self.class.name}: Cannot update message. Wrong number of arguments (expected text or block)")
+          return
+        end
+
+        if text.is_a?(String)
+          robot.update_interactive_message(response_url, text)
+        elsif block.is_a?(Array)
+          robot.update_interactive_block(response_url, block)
+        else
+          Ruboty.logger.warn("#{self.class.name}: Cannot update message. This is not match argument type.")
+          return
+        end
+      end
+
+      def selected_value(block_id, action_id)
```

Comment:
[imo] `Ruboty-slack_socket_mode` で BlockKitフォームの状態を取得するのはこれが初めての機能実装
になるということで、これはとても必要な機能だと思っています！

という反面、取得できる値が `static_select` の場合に限られており、使う側としてもUIのコンポーネントに合わせて使うメソッドを変えないといけないので、使い勝手が少し気になっています。

提案としては。フォーム全体の値をごそっと返せると便利・・という気がしています。が、これはできるかどうかわからないので、私の手元でも少し試してみたいと思います〜

Reply:
フォーム全体ということは
ラジオボタンやマルチセレクトなど他の入力方法も含めてということですよね
またそれらを含む全てを返すようにしたいということはすごくわかります。
今すぐは無理だと思いますが試してみます。

Reply:
仮実装してみました！ `static_select` と `multi_static_select` の2種類に対応したバージョンとして
payload にある `state` から余分なものを削ぎ落としたデータを返す `states` メソッドです。
変数名とかの名付けはかなり雑なので、ご承知おきくださいませ。

```suggestion
      def states
        ret = {}
        @payload.dig('state', 'values').each do |block_id, actions|
          ret[block_id] = {}
          actions.each do |action_id, data|
            ret[block_id][action_id] = begin
              case data['type']
              when 'multi_static_select' then data['selected_options'].map { |selected_option| selected_option['value'] }
              when 'static_select' then data.dig('selected_option', 'value')
              end
            end
          end
        end
        return ret
      end
```

例えば、Handler からは `message.states.dig('block_id', 'action_id')` とすることで使えるようなイメージです。

Reply:
pushしました。
具体的な内容はPRに記載してます